### DISCUSSION TITLE
optimization of mesh made with optimal parts by merging

### DISCUSSION THREAD

**Discussion by slapin** [zeux/meshoptimizer#792] [General]
Hi all!

I have a mesh produced by merging a lot of simple parts (classic modular design). Currently meshopt_simplify() produces no optimization and meshopt_simplyfySloppy() produce unusable result. Is there some way I could preprocess the mesh for it to be better optimized?
Lets imagine my mesh is made of the same part like 100 cubes, i.e. it has 24 vertices with unique attributes and 2400 vertices total. Is it possible to decimate it to reduce vertex count?
Also I have the less generic case of the same cubes but knowing the texture is 9-colors only and whole surface (i.e. quad or triangle) has the same color? I guess in this case reading from texture we can consider vertices having the same texture color having the same UVs? Or is it better to craft special texture with transitional parts and change UVs according to transitions?

**Comment by zeux** [comment]
> Is it possible to decimate it to reduce vertex count?

New version (well, it's only available in master atm but the new version will ship in a week or two) has `meshopt_SimplifyPrune` flag that can simplify meshes like this by removing individual cubes that are small enough. Otherwise it's not really possible to preserve discontinuities in the attributes in the cubes; some enhancements are planned for future versions to allow that as an option but it's not clear atm when this will happen. Of course doing this would distort the geometry of each individual cube by necessity.

**Reply by slapin** [reply]
I found that for single-colored atlas case it helps to use texture colors and replace UV coordinates on all vertices having the same position if colors match.
Thanks for all the help!
Forgot to mention that MeshOptimizer is great tool! Thanks for all the hard work!
